### PR TITLE
Adds Component API doc page

### PR DESCRIPTION
**Pull request by rvsia**
**Description**

Adds documentation for standard components API.

closes https://github.com/data-driven-forms/react-forms/issues/47

### CODE REVIEW

File: packages/react-renderer-demo/src/editor-demo/docs-components/component-api/component-api.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+### FormGroup wrapped components
+
+Basic components are wrapped in formGroup. FormGroup provides a standard API to show labels, helper texts, errors, etc.
+
+All those components provides a shared group of props:
+
+| Prop        | Type           | Description  |
+| ------------- |:-------------:| -----:|
+| label     | node | A label of the field |
```
Comment (Hyperkid123) [inline packages/react-renderer-demo/src/editor-demo/docs-components/component-api/component-api.md]: Is string also covered by node?

File: packages/react-renderer-demo/src/editor-demo/docs-components/component-api/component-api.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+<br/>
+
+#### Checkbox
+
+| Prop        | Type           | Description  |
+| ------------- |:-------------:| -----:|
+| options     | array | Options in format { label: 'Label', value: value }, it will make it multiple a choice list. (optional)|
+
+<div style={{ float: 'right'}}>
+    <Button><Link component={NavLink} to={ `/component-example/checkbox` }>To example</Link></Button>
```
Comment (Hyperkid123) [inline packages/react-renderer-demo/src/editor-demo/docs-components/component-api/component-api.md]: There should be link to both single and multiple chekcbox variant

File: packages/react-renderer-demo/src/editor-demo/docs-components/component-api/component-api.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+| Prop        | Type           | Description  |
+| ------------- |:-------------:| -----:|
+| options     | array | Options in format { label: 'Label', value: value } |
+
+<div style={{ float: 'right'}}>
+    <Button><Link component={NavLink} to={ `/component-example/radio` }>To example</Link></Button>
+</div>
+<br/>
+
+#### Datepicker
```
Comment (Hyperkid123) [inline packages/react-renderer-demo/src/editor-demo/docs-components/component-api/component-api.md]: Just making a note here, we don't have any API specified for PF4 just yet and MUI is probably not the same as for PF3.

Comment (rvsia) [reply packages/react-renderer-demo/src/editor-demo/docs-components/component-api/component-api.md]: Yeah, I know. I just wanted to suggest that we could define some standard API for it and 
eventually re-implement it in other packages. What do you think? Or should I leave it for now just with the `placeholder` prop?

File: packages/react-renderer-demo/src/editor-demo/docs-components/component-api/component-api.md (hunk 1 above)
Comment (rvsia) [reply packages/react-renderer-demo/src/editor-demo/docs-components/component-api/component-api.md]: Should I add `string` to it?

File: packages/react-renderer-demo/src/editor-demo/docs-components/component-api/component-api.md (hunk 3 above)
Comment (Hyperkid123) [reply packages/react-renderer-demo/src/editor-demo/docs-components/component-api/component-api.md]: We will need it so we should think about the API. But there is no clue what kind of component will PF4 come up with.

**Pull request merged**